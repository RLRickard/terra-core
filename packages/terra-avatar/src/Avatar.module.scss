@import '~terra-mixins';

:local {
  .avatar {
    align-items: center;
    border: var(--terra-avatar-border);
    border-radius: 50%;
    box-shadow: var(--terra-avatar-box-shadow);
    color: var(--terra-avatar-color, rgb(155, 159, 161));
    display: inline-flex;
    height: var(--terra-avatar-size, 3.429em);
    justify-content: center;
    line-height: 0;
    overflow: hidden;
    vertical-align: middle; // Fixes gap below image variant
    width: var(--terra-avatar-size, 3.429em);
  }

  .avatar.neutral {
    background-color: var(--terra-avatar-neutral-background-color, rgb(222, 223, 224));
  }

  .avatar.one {
    background-color: var(--terra-avatar-one-background-color, rgb(222, 223, 224));
  }

  .avatar.two {
    background-color: var(--terra-avatar-two-background-color, rgb(222, 223, 224));
  }

  .avatar.three {
    background-color: var(--terra-avatar-three-background-color, rgb(222, 223, 224));
  }

  .avatar.four {
    background-color: var(--terra-avatar-four-background-color, rgb(222, 223, 224));
  }

  .avatar.five {
    background-color: var(--terra-avatar-five-background-color, rgb(222, 223, 224));
  }

  .avatar.six {
    background-color: var(--terra-avatar-six-background-color, rgb(222, 223, 224));
  }

  .avatar.seven {
    background-color: var(--terra-avatar-seven-background-color, rgb(222, 223, 224));
  }

  .avatar.eight {
    background-color: var(--terra-avatar-eight-background-color, rgb(222, 223, 224));
  }

  .avatar.nine {
    background-color: var(--terra-avatar-nine-background-color, rgb(222, 223, 224));
  }

  .avatar.ten {
    background-color: var(--terra-avatar-ten-background-color, rgb(222, 223, 224));
  }

  .avatar.is-deceased {
    opacity: var(--terra-avatar-is-decased-opacity, rgb(222, 223, 224));
  }

  .avatar-icon {
    background-repeat: no-repeat;
    background-size: contain;
    display: block;
    height: 100%;
    width: 100%;

    /* Disabling so that single quotes can be used in svg string */
    /* stylelint-disable string-quotes */
    &.facility {
      background: var(--terra-avatar-icon-facility, inline-svg("<svg data-name='Layer_1' xmlns='http://www.w3.org/2000/svg' fill='rgb(155, 159, 161)' viewBox='0 0 48 48'><path d='M33,20.2V12H15v8.2H9.8V48h12v-9h4.5v9h12V20.2H33z M17.2,42.8h-3V39h3V42.8z M17.2,33.8h-3V30h3V33.8z M25.5,33.8h-3V30h3 V33.8z M29.2,22.5h-3.8v3.8h-3v-3.8h-3.8v-3h3.8v-3.8h3v3.8h3.8V22.5z M33.8,42.8h-3V39h3V42.8z M33.8,33.8h-3V30h3V33.8z'/></svg>"));
    }

    &.user {
      background: var(--terra-avatar-icon-user, inline-svg("<svg data-name='Layer_1' xmlns='http://www.w3.org/2000/svg' fill='rgb(155, 159, 161)' viewBox='0 0 48 48'><path d='M24,8.6c5.1,0,9.3,4.2,9.3,9.3s-4.2,9.3-9.3,9.3s-9.3-4.2-9.3-9.3C14.8,12.8,18.9,8.6,24,8.6z M40,48v-4.7 c0-7.6-6.1-13.8-13.7-13.8c0,0,0,0,0,0h-4.5C14.1,29.6,8,35.7,8,43.3c0,0,0,0,0,0V48H40z'/></svg>"));
    }
    /* stylelint-enable string-quotes */
  }

  .avatar-image {
    height: 100%;
    width: 100%;
  }

  .avatar-text {
    font-size: var(--terra-avatar-font-size, 1.571rem);
  }
}
