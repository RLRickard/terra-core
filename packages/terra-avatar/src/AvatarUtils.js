// const Variants = {
//   AUTO: 'auto',
//   NEUTRAL: 'neutral',
//   ONE: 'one',
//   TWO: 'two',
//   THREE: 'three',
//   FOUR: 'four',
//   FIVE: 'five',
//   SIX: 'six',
//   SEVEN: 'seven',
//   EIGHT: 'eight',
//   NINE: 'nine',
//   TEN: 'ten',
// };
// /**
//  * Map that contains automated color variants.
//  */
// const VariantsMap = new Map([
//   [1, Variants.ONE],
//   [2, Variants.TWO],
//   [3, Variants.THREE],
//   [4, Variants.FOUR],
//   [5, Variants.FIVE],
//   [6, Variants.SIX],
//   [7, Variants.SEVEN],
//   [8, Variants.EIGHT],
//   [9, Variants.NINE],
//   [10, Variants.TEN],
// ]);

/**
 * Map that contains automated color variants.
 */
// const VariantsMap = new Map([
//   [1, 'one'],
//   [2, 'two'],
//   [3, 'three'],
//   [4, 'four'],
//   [5, 'five'],
//   [6, 'six'],
//   [7, 'seven'],
//   [8, 'eight'],
//   [9, 'nine'],
//   [10, 'ten'],
// ]);

const Variants = ['one', 'two', 'three', 'four', 'five', 'six', 'seven', 'eight', 'nine', 'ten'];

// const createVariantsMap = (object) {
//   for( let key in object ) {

//   }
// }

/**
 * Calculates a hash, based on a given string s.
 * @param {*} s
 */
const calculateHash = (s) => {
  let hash = 0;
  let char;

  if (s.trim().length === 0) {
    return hash;
  }

  /* eslint-disable no-bitwise */
  for (let i = 0; i < s.length; i += 1) {
    char = s.charCodeAt(i);
    hash = (hash << 5) - hash + char;
    // Convert to 32bit integer
    hash &= hash;
  }

  // Always return positive hash
  return Math.abs(hash);
};

/**
 *  Returns a color variant, based on a given hash.
 * @param {*} hash
 */
const getVariant = (hash) => {
  const size = Variants.length - 1;
  const index = hash % size;
  return Variants[index];
};


const Utils = {
  Variants,
  // VariantsMap,
  calculateHash,
  getVariant,
};

export default Utils;
