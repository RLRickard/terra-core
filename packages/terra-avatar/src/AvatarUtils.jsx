import React from 'react';
import classNames from 'classnames/bind';
import TerraImage from 'terra-image';
import styles from './Avatar.module.scss';

const cx = classNames.bind(styles);

const AvatarVariants = {
  USER: 'user',
  FACILITY: 'facility',
  MULTIUSER: 'multi-user',
};

const ColorVariants = ['one', 'two', 'three', 'four', 'five', 'six',
  'seven', 'eight', 'nine', 'ten'];

/**
 * Generates a hash, based on a given string s.
 * @param {*} s
 */
const calculateHash = (s) => {
  let hash = 0;
  let char;

  if (s.trim().length === 0) {
    return hash;
  }

  /* eslint-disable no-bitwise */
  for (let i = 0; i < s.length; i += 1) {
    char = s.charCodeAt(i);
    hash = (hash << 5) - hash + char;
    // Convert to 32bit integer
    hash &= hash;
  }

  // Always return positive hash
  return Math.abs(hash);
};

/**
 *  Returns a color variant based on a given hashValue.
 * @param {*} hashValue
 */
const getColorVariant = (hashValue) => {
  const hash = calculateHash(hashValue);
  const size = ColorVariants.length - 1;
  const index = hash % size;
  return ColorVariants[index];
};

/**
 * Returns a color variant based on the following precedence:
 *   1. `color`
 *   2. Hash `hashValue` to generate color variant.
 *   3. Hash `alt`  to generate color variant.
 * @param {*} hashValue
 * @param {*} color
 * @param {*} alt
 */
const setColor = (alt, color, hashValue) => {
  let colorVariant = null;
  if (color !== 'auto') {
    colorVariant = color;
  } else if (hashValue) {
    colorVariant = getColorVariant(hashValue);
  } else {
    colorVariant = getColorVariant(alt);
  }

  return colorVariant;
};

/**
 * Render placeholder.
 * @param {*} alt
 * @param {*} isAriaHidden
 * @param {*} variant
 */
const generateImagePlaceholder = (alt, isAriaHidden, variant) => {
  const avatarIconClassNames = cx(['avatar-icon', variant]);

  return <span className={avatarIconClassNames} role="img" aria-label={alt} alt={alt} aria-hidden={isAriaHidden} />;
};

/**
 * Render image with placeholder.
 * @param {*} image
 * @param {*} alt
 * @param {*} isAriaHidden
 * @param {*} variant
 */
const generateImage = (image, alt, isAriaHidden, variant) => {
  const icon = generateImagePlaceholder(alt, isAriaHidden, variant);

  return <TerraImage className={cx('avatar-image')} src={image} placeholder={icon} alt={alt} />;
};

const Utils = {
  AvatarVariants,
  ColorVariants,
  calculateHash,
  getColorVariant,
  setColor,
  generateImagePlaceholder,
  generateImage,
};

export default Utils;
