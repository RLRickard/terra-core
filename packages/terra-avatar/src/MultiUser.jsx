import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import 'terra-base/lib/baseStyles';
import Utils from './shared/AvatarUtils';
import styles from './shared/Avatar.module.scss';

const cx = classNames.bind(styles);

const propTypes = {
  /**
   * Specifies the alternative text for the image.
   */
  alt: PropTypes.string.isRequired,
  /**
   * Sets the background color. Defaults to `auto`. Color variants are theme specific.
   * Accepted values: `'auto'`, `'neutral'`, `'one'`, `'two'`, `'three'`, `'four'`, `'five'`, `'six'`, `'seven'`, `'eight'`, `'nine'`, `'ten'`.
   */
  color: PropTypes.oneOf(['auto', 'neutral', 'one', 'two', 'three', 'four',
    'five', 'six', 'seven', 'eight', 'nine', 'ten']),
  /**
   * Value used for the hash function when color is set to `auto`. If not provided, hash function utilizes alt.
   */
  hashValue: PropTypes.string,
  /**
   * Overrides the default height.
   */
  height: PropTypes.string,
  /**
   * Whether to hide avatar from the accessiblity API.
   */
  isAriaHidden: PropTypes.bool,
  /**
   * Overrides the default width.
   */
  width: PropTypes.string,
};

const defaultProps = {
  color: 'auto',
  hashValue: undefined,
  height: undefined,
  isAriaHidden: false,
  width: undefined,
};

const MultiUser = ({
  alt,
  color,
  hashValue,
  height,
  isAriaHidden,
  width,
  ...customProps
}) => {
  const colorVariant = Utils.setColor(alt, color, hashValue);
  const attributes = Object.assign({}, customProps);
  const multiUserClassNames = cx([
    'avatar',
    `${colorVariant}`,
    attributes.className,
  ]);

  const multiUserIconClassNames = cx(['icon', Utils.AvatarVariants.MULTIUSER]);
  const multiUserContent = <span className={multiUserIconClassNames} role="img" aria-label={alt} alt={alt} aria-hidden={isAriaHidden} />;

  return (
    <div {...attributes} className={multiUserClassNames} style={{ width, height }}>
      {multiUserContent}
    </div>
  );
};

MultiUser.propTypes = propTypes;
MultiUser.defaultProps = defaultProps;

export default MultiUser;
