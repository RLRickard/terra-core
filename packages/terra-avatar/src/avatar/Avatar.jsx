import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import 'terra-base/lib/baseStyles';
import TerraImage from 'terra-image';
import Utils from '../AvatarUtils';
import styles from '../Avatar.module.scss';

const cx = classNames.bind(styles);

const propTypes = {
  /**
   * Specifies the alternative text for the image.
   */
  alt: PropTypes.string.isRequired,
  /**
   * Sets the background color. Defaults to `auto`.
   * Accepted values: `'auto'`, `'neutral'`, `'one'`, `'two'`, `'three'`, `'four'`, `'five'`, `'six'`, `'seven'`, `'eight'`, `'nine'`, `'ten'`.
   */
  color: PropTypes.oneOf(['auto', 'neutral', 'one', 'two', 'three', 'four',
    'five', 'six', 'seven', 'eight', 'nine', 'ten']),
  /**
   * Value used for the hash function when color is set to `auto`. If not provided, hash function utilizes alt.
   */
  hashValue: PropTypes.string,
  /**
   * Overrides the default height.
   */
  height: PropTypes.string,
  /**
   * The image to display.
   */
  image: PropTypes.string,
  /**
   * One or two letters to display.
   */
  initials: PropTypes.string,
  /**
   * Whether to hide avatar from the accessiblity API.
   */
  isAriaHidden: PropTypes.bool,
  /**
   * Whether the person is desceased.
   */
  isDeceased: PropTypes.bool,
  /**
   * Overrides the default width.
   */
  width: PropTypes.string,
};

const defaultProps = {
  color: 'auto',
  hashValue: undefined,
  height: undefined,
  image: undefined,
  initials: undefined,
  isAriaHidden: false,
  isDeceased: false,
  width: undefined,
};

class Avatar extends React.Component {
  static generateImagePlaceholder(alt, isAriaHidden) {
    const avatarIconClassNames = cx([
      'avatar-icon',
      'user',
    ]);

    return <span className={avatarIconClassNames} role="img" aria-label={alt} alt={alt} aria-hidden={isAriaHidden} />;
  }

  static generateImage(image, alt, isAriaHidden) {
    const icon = Avatar.generateImagePlaceholder(alt, isAriaHidden);

    return <TerraImage className={cx('avatar-image')} src={image} placeholder={icon} alt={alt} />;
  }

  static automateColor(hashValue) {
    const hash = Utils.calculateHash(hashValue);
    // console.log(`hash: ${hash}`);
    const color = Utils.getVariant(hash);
    // console.log(`color: ${color}`);
    return color;
  }

  constructor(props) {
    super(props);

    // If image has been provided we need to generate the image to display and store it in the state
    if (props.image) {
      const { alt, image, isAriaHidden } = props;

      const imageComponent = Avatar.generateImage(image, alt, isAriaHidden);
      this.state = { imageComponent };
    }

    // Checks to run when not in production
    if (process.env.NODE_ENV !== 'production') {
      if (props.image && props.initials) {
        // eslint-disable-next-line
        console.warn('Only one of the props: [image, initials] should be supplied.');
      }
    }
  }

  componentWillReceiveProps(newProps) {
    // If we have an image to display (they take precedence) and one of its attributes have changed
    if (newProps.image && (newProps.image !== this.props.image || newProps.alt !== this.props.alt)) {
      const { alt, image, isAriaHidden } = newProps;

      const imageComponent = Avatar.generateImage(image, alt, isAriaHidden);
      this.setState({ imageComponent });
    }
  }

  render() {
    const {
      alt,
      color,
      hashValue,
      height,
      image,
      initials,
      isAriaHidden,
      isDeceased,
      width,
      ...customProps
    } = this.props;

    const attributes = Object.assign({}, customProps);

    let colorVariant = null;

    if (hashValue) {
      colorVariant = Avatar.automateColor(hashValue);
    } else if (color !== 'auto') {
      colorVariant = color;
    } else {
      colorVariant = Avatar.automateColor(alt);
    }
    console.log(`color variant: ${colorVariant}`);

    const avatarClassNames = cx([
      'avatar',
      `${colorVariant}`,
      attributes.className,
    ]);

    let avatarContent;
    if (image) {
      avatarContent = this.state.imageComponent;
    } else if (initials && (initials.length === 1 || initials.length === 2)) {
      const avatarTextClassNames = cx('avatar-text');

      avatarContent = <span className={avatarTextClassNames} aria-hidden={isAriaHidden}>{initials.toUpperCase()}</span>;
    } else {
      avatarContent = Avatar.generateImagePlaceholder(alt, isAriaHidden);
    }

    return (
      <div {...attributes} aria-label={alt} className={avatarClassNames} style={{ width, height }}>
        {avatarContent}
      </div>
    );
  }
}

Avatar.propTypes = propTypes;
Avatar.defaultProps = defaultProps;

export default Avatar;
