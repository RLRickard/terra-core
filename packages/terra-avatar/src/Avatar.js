import Avatar from './avatar/Avatar';
import Location from './location/Location';
import MultiUser from './multi-user/MultiUser';

const Avatar = {
  Avatar,
  Location,
  MultiUser,
}

export default Avatar;
export { Avatar, Location, MultiUser };
