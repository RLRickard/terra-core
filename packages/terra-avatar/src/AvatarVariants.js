import Avatar from './Avatar';
import Facility from './Facility';
import MultiUser from './MultiUser';

const AvatarVariants = {
  Avatar,
  Facility,
  MultiUser,
};

export default AvatarVariants;
export { Avatar, Facility, MultiUser };
