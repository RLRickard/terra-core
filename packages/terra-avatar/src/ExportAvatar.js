import Avatar from './Avatar';
import Facility from './Facility';
import MultiUser from './MultiUser';

const ExportAvatar = {
  Avatar,
  Facility,
  MultiUser,
};

export default ExportAvatar;
export { Avatar, Facility, MultiUser };
