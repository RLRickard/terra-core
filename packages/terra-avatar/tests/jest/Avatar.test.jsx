import React from 'react';
import { Avatar, Facility, MultiUser } from '../../src/Avatar';
import exampleProfilePhoto from '../../src/terra-dev-site/assets/150x150.jpg';

describe('Avatar', () => {
  // Snapshot Tests
  it('should render a default avatar', () => {
    const avatar = <Avatar alt="user" />;
    const wrapper = render(avatar);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render one initial avatar when one initial is passed in', () => {
    const avatar = <Avatar initials="J" alt="user" />;
    const wrapper = render(avatar);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render two initials avatar when two initials are passed in', () => {
    const avatar = <Avatar initials="JS" alt="user" />;
    const wrapper = render(avatar);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render fallback user avatar when more than two initials are passed in', () => {
    const avatar = <Avatar initials="JJJ" alt="user" />;
    const wrapper = render(avatar);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render image avatar when image is passed in', () => {
    const avatar = <Avatar image={exampleProfilePhoto} alt="placeholder" />;
    const wrapper = render(avatar);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render fallback user avatar when invalid image is passed in', () => {
    const avatar = <Avatar image="https://path/to/invalid_image.jpg" alt="placeholder" />;
    const wrapper = render(avatar);
    expect(wrapper).toMatchSnapshot();
  });

  it('should display warning when both image and initials props are provided', () => {
    const avatar = <Avatar image="https://path/to/invalid_image.jpg" alt="placeholder" initials="JS" />;
    try {
      render(avatar);
    } catch (object) {
      expect(object.message).toContain('Only one of the props: [image, initials] should be supplied.');
    }
  });


  it('should render an avatar with provided height and width', () => {
    const avatar = <Avatar alt="multi-user" height="5em" width="5em" />;
    const wrapper = render(avatar);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render custom props', () => {
    const avatar = <Avatar alt="user" style={{ color: 'purple' }} />;
    const wrapper = render(avatar);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('Facility', () => {
  it('should render a default facility avatar', () => {
    const avatar = <Facility alt="facility" />;
    const wrapper = render(avatar);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render fallback facility avatar when more than two initials are passed in', () => {
    const avatar = <Facility initials="JSP" alt="facility" />;
    const wrapper = render(avatar);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render fallback facility avatar when invalid image is passed in', () => {
    const avatar = <Facility image="https://path/to/invalid_image.jpg" alt="placeholder" />;
    const wrapper = render(avatar);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with a facility avatar with provided height and width', () => {
    const avatar = <Facility alt="multi-user" height="5em" width="5em" />;
    const wrapper = render(avatar);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a facility avatar with custom props', () => {
    const avatar = <Facility alt="facility" style={{ color: 'purple' }} />;
    const wrapper = render(avatar);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('MultiUser', () => {
  it('should render a default multi-user avatar', () => {
    const avatar = <MultiUser alt="multi-user" />;
    const wrapper = render(avatar);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with a multi user with provided height and width', () => {
    const avatar = <MultiUser alt="multi-user" height="5em" width="5em" />;
    const wrapper = render(avatar);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a multi user avatar with custom props', () => {
    const avatar = <MultiUser alt="multi-user" style={{ color: 'purple' }} />;
    const wrapper = render(avatar);
    expect(wrapper).toMatchSnapshot();
  });
});
