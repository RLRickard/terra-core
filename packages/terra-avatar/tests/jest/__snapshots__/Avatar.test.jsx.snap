// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar should render a default avatar 1`] = `
<div
  alt="user"
  class="avatar one"
>
  <span
    alt="user"
    aria-hidden="false"
    aria-label="user"
    class="avatar-icon user"
    role="img"
  />
</div>
`;

exports[`Avatar should render an avatar with provided height and width 1`] = `
<div
  alt="multi-user"
  class="avatar five"
  style="width:5em;height:5em"
>
  <span
    alt="multi-user"
    aria-hidden="false"
    aria-label="multi-user"
    class="avatar-icon user"
    role="img"
  />
</div>
`;

exports[`Avatar should render custom props 1`] = `
<div
  alt="user"
  class="avatar one"
>
  <span
    alt="user"
    aria-hidden="false"
    aria-label="user"
    class="avatar-icon user"
    role="img"
  />
</div>
`;

exports[`Avatar should render fallback user avatar when invalid image is passed in 1`] = `
<div
  alt="placeholder"
  class="avatar five"
>
  <div>
    <div
      class="hidden"
    >
      <img
        alt="placeholder"
        class="image default avatar-image"
        src="https://path/to/invalid_image.jpg"
      />
    </div>
    <div>
      <span
        alt="placeholder"
        aria-hidden="false"
        aria-label="placeholder"
        class="avatar-icon user"
        role="img"
      />
    </div>
  </div>
</div>
`;

exports[`Avatar should render fallback user avatar when more than two initials are passed in 1`] = `
<div
  alt="user"
  class="avatar one"
>
  <span
    alt="user"
    aria-hidden="false"
    aria-label="user"
    class="avatar-icon user"
    role="img"
  />
</div>
`;

exports[`Avatar should render image avatar when image is passed in 1`] = `
<div
  alt="placeholder"
  class="avatar five"
>
  <div>
    <div
      class="hidden"
    >
      <img
        alt="placeholder"
        class="image default avatar-image"
        src="test-file-stub"
      />
    </div>
    <div>
      <span
        alt="placeholder"
        aria-hidden="false"
        aria-label="placeholder"
        class="avatar-icon user"
        role="img"
      />
    </div>
  </div>
</div>
`;

exports[`Avatar should render one initial avatar when one initial is passed in 1`] = `
<div
  alt="user"
  class="avatar one"
>
  <span
    aria-hidden="false"
    class="avatar-text"
  >
    J
  </span>
</div>
`;

exports[`Avatar should render two initials avatar when two initials are passed in 1`] = `
<div
  alt="user"
  class="avatar one"
>
  <span
    aria-hidden="false"
    class="avatar-text"
  >
    JS
  </span>
</div>
`;

exports[`Facility should render a default facility avatar 1`] = `
<div
  alt="facility"
  class="avatar nine"
>
  <span
    alt="facility"
    aria-hidden="false"
    aria-label="facility"
    class="avatar-icon facility"
    role="img"
  />
</div>
`;

exports[`Facility should render a facility avatar with custom props 1`] = `
<div
  alt="facility"
  class="avatar nine"
>
  <span
    alt="facility"
    aria-hidden="false"
    aria-label="facility"
    class="avatar-icon facility"
    role="img"
  />
</div>
`;

exports[`Facility should render fallback facility avatar when invalid image is passed in 1`] = `
<div
  alt="placeholder"
  class="avatar five"
>
  <div>
    <div
      class="hidden"
    >
      <img
        alt="placeholder"
        class="image default avatar-image"
        src="https://path/to/invalid_image.jpg"
      />
    </div>
    <div>
      <span
        alt="placeholder"
        aria-hidden="false"
        aria-label="placeholder"
        class="avatar-icon facility"
        role="img"
      />
    </div>
  </div>
</div>
`;

exports[`Facility should render fallback facility avatar when more than two initials are passed in 1`] = `
<div
  alt="facility"
  class="avatar nine"
  initials="JSP"
>
  <span
    alt="facility"
    aria-hidden="false"
    aria-label="facility"
    class="avatar-icon facility"
    role="img"
  />
</div>
`;

exports[`Facility should render with a facility avatar with provided height and width 1`] = `
<div
  alt="multi-user"
  class="avatar five"
  style="width:5em;height:5em"
>
  <span
    alt="multi-user"
    aria-hidden="false"
    aria-label="multi-user"
    class="avatar-icon facility"
    role="img"
  />
</div>
`;

exports[`MultiUser should render a default multi-user avatar 1`] = `
<div
  alt="multi-user"
  class="avatar five"
>
  <span
    alt="multi-user"
    aria-hidden="false"
    aria-label="multi-user"
    class="avatar-icon multi-user"
    role="img"
  />
</div>
`;

exports[`MultiUser should render a multi user avatar with custom props 1`] = `
<div
  alt="multi-user"
  class="avatar five"
>
  <span
    alt="multi-user"
    aria-hidden="false"
    aria-label="multi-user"
    class="avatar-icon multi-user"
    role="img"
  />
</div>
`;

exports[`MultiUser should render with a multi user with provided height and width 1`] = `
<div
  alt="multi-user"
  class="avatar five"
  style="width:5em;height:5em"
>
  <span
    alt="multi-user"
    aria-hidden="false"
    aria-label="multi-user"
    class="avatar-icon multi-user"
    role="img"
  />
</div>
`;
